src/spikeinterface/core/tests/test_waveform_tools.py

```diff
@@ -52,96 +54,88 @@ def test_waveform_tools():
     unit_ids = sorting.unit_ids
 
     some_job_kwargs = [
-        {},
         {"n_jobs": 1, "chunk_size": 3000, "progress_bar": True},
         {"n_jobs": 2, "chunk_size": 3000, "progress_bar": True},
     ]
+    some_modes = [
+        {"mode": "memmap"},
+    ]
+    if platform.system() != "Windows":
+        # shared memory on windows is buggy...
+        some_modes.append(
+            {
+                "mode": "shared_memory",
+            }
+        )
+
+    some_sparsity = [
+        dict(sparsity_mask=None),
+        dict(sparsity_mask=np.random.randint(0, 2, size=(unit_ids.size, recording.channel_ids.size), dtype="bool")),
+    ]
 
     # memmap mode
-    list_wfs = []
+    list_wfs_dense = []
+    list_wfs_sparse = []
     for j, job_kwargs in enumerate(some_job_kwargs):
```

Comment:
As an another story against single variable names. 

I want to know why are you using j, k, l and here, what do I do?

I have to inspect and look for those three letters in all the linew that you are added. I can't do search on the term because is a single letter. I takes me a couple of minutes to realize that they are used in line 85 to name the file and that seems (but I am not sure) that is the only use.

If you use something explicit like mode_kwargs_index, sparsity_kwargs_index then I just ctrl+f and the term and I am done. 

The benefit of single letter variables? an aesthetic preference

Reply:
For the general case you are right.
But nested loop with i, j, k, l is a classic. no ?
I think really like to have single letter variable name for counter.
For this simple test this is OK

Reply:
Finally I remove the folder kwargs and put file_path instead which is even better the name is external.
We will need this in several context.

Reply:
Sorry, Sam, I should have written this on my diary and no in your PR.

[You are not alone
](https://www.reddit.com/r/AskProgramming/comments/e9sfds/what_are_the_acceptable_uses_for_single_letter/)
Lots of people love using single letter variables for counter and will die on that hill. 

That said, I have never seen any argument for keeping the single letter convention other than aestethical preference and I was just pointing out costs by means of a specific story. In this case because the scope of the function is a little too large and it was not obvious where they are used.